src/plugins/workspace/public/components/workspace_form/utils.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DEFAULT_CHECKED_FEATURES_IDS } from '../../../common/constants';
+
+import { WorkspaceFeature, WorkspaceFeatureGroup, WorkspaceFormErrors } from './types';
+
+export const isWorkspaceFeatureGroup = (
+  featureOrGroup: WorkspaceFeature | WorkspaceFeatureGroup
+): featureOrGroup is WorkspaceFeatureGroup => 'features' in featureOrGroup;
+
+export const isDefaultCheckedFeatureId = (id: string) => {
+  return DEFAULT_CHECKED_FEATURES_IDS.indexOf(id) > -1;
+};
+
+export const appendDefaultFeatureIds = (ids: string[]) => {
+  // concat default checked ids and unique the result
+  return Array.from(new Set(ids.concat(DEFAULT_CHECKED_FEATURES_IDS)));
+};
+
+export const isValidNameOrDescription = (input?: string) => {
+  if (!input) {
+    return true;
+  }
+  const regex = /^[0-9a-zA-Z()_\[\]\-\s]+$/;
+  return regex.test(input);
+};
+
+export const getNumberOfErrors = (formErrors: WorkspaceFormErrors) => {
```

Comment:
is this more simply ?

like as  return Object.keys(formErrors).length;

Reply:
The permissions field was be removed in this PR. The permissions field may includes multiple errors. We can't just list the keys of formErrors. Do you think it should be refactor to `Array.prototype.reduce`?